### PR TITLE
Seed debt

### PR DESCRIPTION
**Pull request by DominikPanic**

### CODE REVIEW

**Comment by Zostroll** [comment]
This removes getting seeds by taking a sample with wirecutters or a knife.

**Comment by DominikPanic** [comment]
> 
> 
> This removes getting seeds by taking a sample with wirecutters or a knife.

I never ever saw the cuttings used, but I guess I can add that back.

**Comment by Shadow-Quill** [comment]
Muh description

File: code/modules/hydroponics/seed_datums.dm [hunk 1]
```diff
@@ -33,7 +28,14 @@ proc/populate_seed_list()
 
 		used_masks += gene_mask
 		gene_tags -= gene_tag
-		gene_tag_masks[gene_tag] = gene_mask
+		GLOB.gene_tag_masks[gene_tag] = gene_mask
+
+	// Make sure any seed packets that were mapped in are updated
+	// correctly (since the seed datums did not exist a tick ago).
+	for(var/obj/item/seeds/S in GLOB.item_list)
+		S.update_seed()
```
Comment (ninjanomnom) [inline code/modules/hydroponics/seed_datums.dm]: Just have the seed list get initialized before atoms are initialized

File: code/game/machinery/vending/vending.dm [hunk 2]
```diff
@@ -107,7 +107,7 @@
 		var/price = prices[typepath]
 		if(isnull(amount)) amount = 1
 
-		var/obj/item/temp_path = typepath
+		var/obj/item/temp_path = new typepath
```
Comment (ninjanomnom) [inline code/game/machinery/vending/vending.dm]: Stop instancing things that dont need to be instanced

File: code/modules/hydroponics/hydro_tray.dm [hunk 3]
```diff
@@ -503,8 +503,8 @@
 
 	var/previous_plant = seed.display_name
 	var/newseed = seed.get_mutant_variant()
-	if(newseed in seed_types)
-		seed = seed_types[newseed]
+	if(newseed in GLOB.seed_types)
+		seed = GLOB.seed_types[newseed]
```
Comment (ninjanomnom) [inline code/modules/hydroponics/hydro_tray.dm]: The if check is unneeded here

```suggestion
		seed = GLOB.seed_types[newseed] || seed
```

File: code/game/machinery/vending/vending.dm (hunk 2 above)
Comment (DominikPanic) [reply code/game/machinery/vending/vending.dm]: It's either instancing or snowflake code that handles seeds specifically :/

**Pull request merged**